lms/djangoapps/teams/views.py

```diff
@@ -485,8 +487,20 @@ def has_object_permission(self, request, view, obj):
             """Returns true if the user is enrolled or is staff."""
             return has_team_api_access(request.user, obj.course_id)
 
+    class IsStaffOrTAOrReadOnly(IsStaffOrReadOnly):
+        """Permission that checks to see if the user is global staff, course
+        staff, or has discussion privileges. If none of those conditions are
+        met, only read access will be granted.
+        """
+
+        def has_object_permission(self, request, view, obj):
+            return (
+                has_discussion_privileges(request.user, obj.course_id) or
+                IsStaffOrReadOnly.has_object_permission(self, request, view, obj)
```

Comment:
I'm having a hard time getting this to work (admittedly - it could just be a symptom of my Python noobity)

When using super(<ChildClass>, self).has_object_permission(...) like you specified, I get the following: NameError: global name 'IsStaffOrPrivilegedOrReadOnly' is not defined

When I use super(<BaseClass>, self).has_object_permission(...), test_access_4___student_enrolled___403_ in TestUpdateTeamAPI fails with a 200 instead of the expected 403.

<BaseClass>.has_object_permission(self, ...) seems to be the only variation that passes all tests. Any suggestions?